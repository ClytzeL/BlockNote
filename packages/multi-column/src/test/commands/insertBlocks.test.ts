import { describe, expect, it } from "vitest";

import { setupTestEnv } from "../setupTestEnv.js";

const getEditor = setupTestEnv();

describe("Test insertBlocks", () => {
  it("Insert empty column list", () => {
    getEditor().insertBlocks([{ type: "columnList" }], "paragraph-0", "after");

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Insert column list with empty column", () => {
    getEditor().insertBlocks(
      [
        {
          type: "columnList",
          children: [
            {
              type: "column",
            },
          ],
        },
      ],
      "paragraph-0",
      "after"
    );

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Insert column list with paragraph", () => {
    getEditor().insertBlocks(
      [
        {
          type: "columnList",
          children: [
            {
              type: "column",
              children: [
                {
                  type: "paragraph",
                  content: "Inserted Column Paragraph",
                },
              ],
            },
          ],
        },
      ],
      "paragraph-0",
      "after"
    );

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Insert empty column into column list", () => {
    getEditor().insertBlocks(
      [
        {
          type: "column",
        },
      ],
      "column-0",
      "before"
    );

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Insert column with paragraph into column list", () => {
    getEditor().insertBlocks(
      [
        {
          type: "column",
          children: [
            {
              type: "paragraph",
              content: "Inserted Column Paragraph",
            },
          ],
        },
      ],
      "column-0",
      "before"
    );

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Insert column list into paragraph", () => {
    getEditor().insertBlocks(
      [
        {
          type: "columnList",
          children: [
            {
              type: "column",
              children: [
                {
                  type: "paragraph",
                  content: "Inserted Column Paragraph",
                },
              ],
            },
          ],
        },
      ],
      "nested-paragraph-0",
      "after"
    );

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Insert column into paragraph", () => {
    getEditor().insertBlocks(
      [
        {
          type: "column",
          children: [
            {
              type: "paragraph",
              content: "Inserted Column Paragraph",
            },
          ],
        },
      ],
      "nested-paragraph-0",
      "after"
    );

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Insert paragraph into column list", () => {
    getEditor().insertBlocks(
      [
        {
          type: "paragraph",
          content: "Inserted Column List Paragraph",
        },
      ],
      "column-0",
      "after"
    );

    expect(getEditor().document).toMatchSnapshot();
  });

  it("Insert paragraph into column", () => {
    getEditor().insertBlocks(
      [
        {
          type: "paragraph",
          content: "Inserted Column List Paragraph",
        },
      ],
      "column-paragraph-0",
      "after"
    );

    expect(getEditor().document).toMatchSnapshot();
  });
});
