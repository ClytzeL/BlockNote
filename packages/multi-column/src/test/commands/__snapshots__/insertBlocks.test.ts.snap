// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test insertBlocks > Insert column list with empty column 1`] = `
[
  {
    "children": [
      {
        "children": [],
        "content": [
          {
            "styles": {},
            "text": "Nested Paragraph 0",
            "type": "text",
          },
        ],
        "id": "nested-paragraph-0",
        "props": {
          "backgroundColor": "default",
          "textAlignment": "left",
          "textColor": "default",
        },
        "type": "paragraph",
      },
    ],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 0",
        "type": "text",
      },
    ],
    "id": "paragraph-0",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [
      {
        "children": [],
        "content": undefined,
        "id": "1",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
    ],
    "content": undefined,
    "id": "0",
    "props": {},
    "type": "columnList",
  },
  {
    "children": [
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 0",
                "type": "text",
              },
            ],
            "id": "column-paragraph-0",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 1",
                "type": "text",
              },
            ],
            "id": "column-paragraph-1",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "column-0",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 2",
                "type": "text",
              },
            ],
            "id": "column-paragraph-2",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 3",
                "type": "text",
              },
            ],
            "id": "column-paragraph-3",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "column-1",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
    ],
    "content": undefined,
    "id": "column-list-0",
    "props": {},
    "type": "columnList",
  },
  {
    "children": [],
    "content": [],
    "id": "trailing-paragraph",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
]
`;

exports[`Test insertBlocks > Insert empty column list 1`] = `
[
  {
    "children": [
      {
        "children": [],
        "content": [
          {
            "styles": {},
            "text": "Nested Paragraph 0",
            "type": "text",
          },
        ],
        "id": "nested-paragraph-0",
        "props": {
          "backgroundColor": "default",
          "textAlignment": "left",
          "textColor": "default",
        },
        "type": "paragraph",
      },
    ],
    "content": [
      {
        "styles": {},
        "text": "Paragraph 0",
        "type": "text",
      },
    ],
    "id": "paragraph-0",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
  {
    "children": [],
    "content": undefined,
    "id": "0",
    "props": {},
    "type": "columnList",
  },
  {
    "children": [
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 0",
                "type": "text",
              },
            ],
            "id": "column-paragraph-0",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 1",
                "type": "text",
              },
            ],
            "id": "column-paragraph-1",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "column-0",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
      {
        "children": [
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 2",
                "type": "text",
              },
            ],
            "id": "column-paragraph-2",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
          {
            "children": [],
            "content": [
              {
                "styles": {},
                "text": "Column Paragraph 3",
                "type": "text",
              },
            ],
            "id": "column-paragraph-3",
            "props": {
              "backgroundColor": "default",
              "textAlignment": "left",
              "textColor": "default",
            },
            "type": "paragraph",
          },
        ],
        "content": undefined,
        "id": "column-1",
        "props": {
          "width": 1,
        },
        "type": "column",
      },
    ],
    "content": undefined,
    "id": "column-list-0",
    "props": {},
    "type": "columnList",
  },
  {
    "children": [],
    "content": [],
    "id": "trailing-paragraph",
    "props": {
      "backgroundColor": "default",
      "textAlignment": "left",
      "textColor": "default",
    },
    "type": "paragraph",
  },
]
`;
